src/Solvers/solve_for_pressure.jl

```diff
@@ -21,6 +21,25 @@ function solve_for_pressure!(pressure, solver, arch, grid, Δt, U★)
     return nothing
 end
 
+function solve_for_pressure!(pressure, solver::PreconditionedConjugateGradientSolver, arch, grid, Δt, U★)
+    RHS = solver.settings.RHS
+    rhs_event = launch!(arch, grid, :xyz,
+                        calculate_pressure_right_hand_side!, RHS, arch, grid, Δt, U★,
+                        dependencies = Event(device(arch)))
+    wait(device(arch), rhs_event)
+    fill_halo_regions!(RHS, solver.settings.bcs, arch, grid)
+
+    x = solver.settings.x
+    x .= 0
+
+    solve_poisson_equation!(solver, RHS, x)
+
+    fill_halo_regions!(x, solver.settings.bcs, arch, grid)
+    pressure.data .= x
+
+    return nothing
+end
```

Comment:
Might want to nuke this too?

Reply:
Do we anticipate adding a regression test that uses the PCG solver + incompressible model? If so then I think we need this function.

But it could be defined in the regression test script if we don't want users using PCG solver + incompressible model.

Reply:
@ali-ramadhan we may want a PCG within rigid lid capability one day. Not sure anyone wants it at moment. We could tidy this now and then merge without squashing when ready, so we have everything recorded in PR for future use as needed. 

Some regression test would not be bad to keep in if it can be, so we know when things break. If that is a pain though we could leave for later.

Reply:
Sounds good, I'll nuke it for now and we can always fish it out of git history.